### PR TITLE
Service Registration: Document State

### PR DESCRIPTION
**Pull request by kmh287**
This pull request distinguishes installing/registering the document state service and creating it.Please see #7839 for more context.

### CODE REVIEW

File: src/service/document-state.js [hunk 1]
```diff
@@ -148,8 +147,16 @@ export class DocumentState {
 
 /**
  * @param {!Window} window
+ */
+export function installDocumentState(window) {
```
Comment (choumx) [inline src/service/document-state.js]: `installDocumentStateService`

File: src/service/document-state.js [hunk 2]
```diff
@@ -148,8 +147,16 @@ export class DocumentState {
...
+  registerServiceBuilder(window, 'documentState', DocumentState);
+}
+
+/**
+ * @param {!Window} window
  * @return {!DocumentState}
  */
 export function documentStateFor(window) {
-  return fromClass(window, 'documentState', DocumentState);
+  installDocumentState(window);
```
Comment (choumx) [inline src/service/document-state.js]: Doesn't installation typically happen separately?

File: src/service/document-state.js (hunk 1 above)
Comment (kmh287) [inline src/service/document-state.js]: Done.

File: src/service/document-state.js (hunk 2 above)
Comment (kmh287) [inline src/service/document-state.js]: all callers assign the result to a value and operate on it immediately.

**Comment by kmh287** [comment]
@choumx Good to go?

**Pull request merged**